repose-aggregator/components/services/http-logging-service/http-logging-service-impl/src/main/scala/org/openrepose/core/services/httplogging/HttpLoggingServiceImpl.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.core.services.httplogging
+
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.annotation.{PostConstruct, PreDestroy}
+import javax.inject.{Inject, Named}
+import org.jtwig.JtwigModel
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.httplogging.config.HttpLoggingConfig
+
+import scala.collection.JavaConverters._
+
+/**
+  * See [[HttpLoggingService]] for details.
+  */
+@Named
+class HttpLoggingServiceImpl @Inject()(configurationService: ConfigurationService,
+                                       httpLoggingConfigListener: HttpLoggingConfigListener)
+  extends HttpLoggingService with StrictLogging {
+
+  import HttpLoggingServiceImpl._
+
+  @PostConstruct
+  def init(): Unit = {
+    logger.trace("Initializing HTTP Logging Service")
+
+    val defaultConfigSchemaUrl = classOf[HttpLoggingServiceImpl].getResource(DefaultConfigSchema)
+    configurationService.subscribeTo(
+      DefaultConfig,
+      defaultConfigSchemaUrl,
+      httpLoggingConfigListener,
+      classOf[HttpLoggingConfig])
+
+    logger.trace("Initialized HTTP Logging Service")
+  }
+
+  @PreDestroy
+  def destroy(): Unit = {
+    logger.trace("Destroying HTTP Logging Service")
+
+    configurationService.unsubscribeFrom(DefaultConfig, httpLoggingConfigListener)
+
+    logger.trace("Destroyed HTTP Logging Service")
+  }
+
+  override def open(): HttpLoggingContext = {
+    val httpLoggingContext = new HttpLoggingContext()
+    logger.trace("Opened a new HTTP logging context {}", s"${httpLoggingContext.hashCode()}")
+    httpLoggingContext
+  }
+
+  override def close(httpLoggingContext: HttpLoggingContext): Unit = {
```

Comment:
I did. Doing so was an immediate solution to the problem with Jetty recycling the request object, though there were other options to solve that problem. Even so, the async bit made calling this method unpredictable, as async processing is, so I removed it.

The key points here are:

- Even without the async processing in this method, we should get away without adding any overhead to the response time. The reason is that the Jetty hook won't call this until the response has already been sent to the requester. That being said, with this approach we do tie up a Jetty processing thread rather than a thread from our own thread pool, but I don't suspect that will have much of an impact. If it does, we can always address it later.
- If a user of this API desires async processing, they can always implement it on their end. This way the user can decide whether or not to log asynchronously.